### PR TITLE
[New Rule] Potential Masquerading as Windows System32 DLL

### PR DESCRIPTION
**Pull request by w0rk3r**
## Issue

#2938 

## Summary

Identifies suspicious instances of default system32 DLLs either unsigned or signed with non-MS certificates. This can potentially indicate the attempt to masquerade as system DLLs, perform DLL Search Order Hijacking or backdoor and resign legitimate DLLs.

### A bit about the Process

To identify the default DLLs in different versions of Windows, I've collected all the files under System32 in two ways: Non-recursive and recursive by using https://learn.microsoft.com/en-us/sysinternals/downloads/sigcheck. Then I parsed the sigcheck output into JSON and used Python to get which DLLs are present in 2 or more versions and which subject names the certificates have.

The data is published here: https://github.com/w0rk3r/WindowsSignerInfo

### A bit about the Problem

The result is huge: ~3200 DLL names, so I have a concern about performance problems, and I would like your input to decide whether to keep this rule as an individual rule or break it up into several rules (Or any other idea we could explore).

I chose to do the PR as an individual rule right now because it is a BBR rule and can be monitored closely after the release and can be easily broken down into several rules, but not the other way around.

### CODE REVIEW

File: rules_building_block/defense_evasion_masquerading_windows_dll.toml [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+Identifies suspicious instances of default system32 DLLs either unsigned or signed with non-MS certificates. This can
+potentially indicate the attempt to masquerade as system DLLs, perform DLL Search Order Hijacking or backdoor and resign
+legitimate DLLs.
+"""
+from = "now-119m"
+interval = "60m"
+index = ["logs-endpoint.events.*"]
+language = "eql"
+license = "Elastic License v2"
+name = "Potential Masquerading as Windows System32 DLL"
```
Comment (Aegrah) [inline rules_building_block/defense_evasion_masquerading_windows_dll.toml:21]:
```suggestion
name = "Potential Masquerading as System32 DLL"
```

File: rules_building_block/defense_evasion_masquerading_windows_dll.toml [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+risk_score = 21
+rule_id = "fb01d790-9f74-4e76-97dd-b4b0f7bf6435"
+severity = "low"
+tags = ["Domain: Endpoint", "OS: Windows", "Use Case: Threat Detection", "Tactic: Defense Evasion", "Rule Type: BBR"]
```
Comment (Aegrah) [inline rules_building_block/defense_evasion_masquerading_windows_dll.toml:25]:
```suggestion
tags = ["Domain: Endpoint", "Data Source: Elastic Defend", "OS: Windows", "Use Case: Threat Detection", "Tactic: Defense Evasion", "Rule Type: BBR"]
```

**Review by Aegrah — APPROVED** [review]
LGTM. Good choice of BBR.

**Comment by w0rk3r** [comment]
After talking with @Samirbous, I've removed more complex file names, such as longer dll names (>=20 chars) or more than two dots in the name to enhance performance by reducing the terms in the query, which reduced it from 3391 to 2615 (-22.8%).

I also changed the logic to exclude some system paths, as those require TrustedInstaller privileges to modify.

The old query contained a `dll.code_signature.status : "*"` condition, which excluded unsigned matches, which wasn't intended.

**Review by Samirbous — APPROVED** [review]

**Pull request merged**